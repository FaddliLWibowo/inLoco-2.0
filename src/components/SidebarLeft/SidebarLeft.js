import React from 'react';
import MenuHeader from '../MenuHeader/MenuHeader.js';
import MenuContainer from '../Menu/MenuContainer.js';
import SearchLayer from '../SearchLayer/SearchLayer.js';
import Measure from 'react-measure';

const SidebarLeft = ({searchLayer, showMenu, hideMenu}) => {
    var cssClass = 'sidebar-left allow-transition ';
    console.log(showMenu);
    if (!showMenu) {
        cssClass += 'hide-sidebar';
    }
    return (
        <Measure>
            {({width, height}) =>
                <div className={cssClass}>
                    <MenuHeader hideMenu={hideMenu}/>
                    <MenuContainer sidebarLeftWidth={width} sidebarLeftHeight={height}/>
                    <SearchLayer searchLayer={searchLayer}/>
                </div>
            }
        </Measure>
    );
};

export default SidebarLeft;
