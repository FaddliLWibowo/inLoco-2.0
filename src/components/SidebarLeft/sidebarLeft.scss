.sidebar-left {
    background: $menu-bg-color;
    position: fixed;
    top: 0;
    left: 0;
    width: 300px;
    height: 100%;
    overflow-y: auto;
    z-index: 1001;

    &--logo {
        height: 43px;
        margin: 20px 0 0 15px;
        outline: 1px solid red;
        width: 195px;

    }

    &--title {
        color: $sidebar-left-highlight-color;
        font: normal 16px sans-serif;
        margin: 20px 0 15px 15px;
        text-transform: uppercase;
    }

    &--caption {
        color: $menu-text-color;
        display: block;
        font-size: 80%;
        text-transform: initial;
    }
}
