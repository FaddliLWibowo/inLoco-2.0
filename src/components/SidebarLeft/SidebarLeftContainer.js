import React from 'react';
import SidebarLeft from './SidebarLeft';
import { connect } from 'react-redux';

const mapDispatchToProps = (dispatch) => {
    return {
        searchLayer: (text) => {
            dispatch({
                type: 'SEARCH_LAYER',
                text
            })
        }
    };
};

const SidebarLeftContainer = connect(
    null,
    mapDispatchToProps
)(SidebarLeft)

export default SidebarLeftContainer;
