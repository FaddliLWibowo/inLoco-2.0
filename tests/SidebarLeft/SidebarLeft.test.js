import React from 'react';
import SidebarLeft from '../../src/components/SidebarLeft/SidebarLeft.js';
import renderer from 'react-test-renderer';

it('SidebarLeft component renders correctly', () => {
    const sidebarLeft  = shallow(<SidebarLeft/>);
    expect(sidebarLeft).toMatchSnapshot();
});
