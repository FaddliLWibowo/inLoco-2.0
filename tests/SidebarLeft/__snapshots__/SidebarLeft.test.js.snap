// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarLeft component renders correctly with no data 1`] = `<_class />`;

exports[`SidebarLeft component renders correctly with real data -> menu open 1`] = `<_class />`;
